src/client.rs

```diff
@@ -303,6 +304,13 @@ impl RequestBuilder {
 
                 url = match loc {
                     Ok(loc) => {
+
+                        if loc.domain() == url.domain() {
+                            if let Some(&SetCookie(ref cookies)) = res.headers.get::<SetCookie>() {
+                                headers.set(Cookie(cookies.iter().map(|c| c.to_owned()).collect::<Vec<String>>()));
```

Comment:
This should mean that the sent `Cookie` headers will contain all the attributes from the `Set-Cookie` header, aka we'd be sending `Cookie: foo=bar; Path=/; Domain=foo.bar; Expires=123`, when it should just be `Cookie: foo=bar`.

Additionally, we could be sending cookies incorrectly. The `Path` and `Domain` of the set cookie could possibly not match the new location.

All this logic is tricky to get correct, but it is a goal of reqwest to do it. See #14.